src/install.c

```diff
@@ -499,6 +499,10 @@ static void prepare_environment(GSubprocessLauncher *launcher, gchar *update_sou
 			for (GList *l = manifest->images; l != NULL; l = l->next) {
 				RaucImage *img = l->data;
 				if (g_str_equal(slot->sclass, img->slotclass)) {
+					/* If no filename is set, do not attempt to set environment variables */
+					if (!img->filename)
+						break;
```

Comment:
Shouldn't this be `continue`? Also, the slotclass is still relevant.

Reply:
Yes, but today's me says: There should be no break/continue at all. We have several other places in the code that unconditionally set these env variables but can deal with C variables being NULL. Doing it the same way here seems to be the most consistent way for me.